src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php

```diff
@@ -285,7 +285,18 @@ protected function cancelCookie(Request $request)
             $this->logger->debug(sprintf('Clearing remember-me cookie "%s"', $this->options['name']));
         }
 
-        $request->attributes->set(self::COOKIE_ATTR_NAME, new Cookie($this->options['name'], null, 1, $this->options['path'], $this->options['domain']));
+        // Hard coded the default values for secure and http only, would be better if those where constants
+        $request->attributes->set(
+            self::COOKIE_ATTR_NAME,
+            new Cookie($this->options['name'],
+                null,
+                1,
+                $this->options['path'],
+                $this->options['domain'],
+                isset($this->options['secure']) ? $this->options['secure'] : false,
```

Comment:
What about using `array_merge` in constructor to be sure that this options always exists?